docs/attributes-registry/process.md

```diff
@@ -11,12 +11,31 @@
+| `process.end` | string | The date and time the process ended, in ISO 8601 format. | `2023-11-21T09:26:12.315Z` |
+| `process.env_vars` | string[] | Array of environment variable bindings. [1] | `[PATH=/usr/local/bin;/usr/bin, USER=ubuntu]` |
 | `process.executable.name` | string | The name of the process executable. On Linux based systems, can be set to the `Name` in `proc/[pid]/status`. On Windows, can be set to the base name of `GetProcessImageFileNameW`. | `otelcol` |
 | `process.executable.path` | string | The full path to the process executable. On Linux based systems, can be set to the target of `proc/[pid]/exe`. On Windows, can be set to the result of `GetProcessImageFileNameW`. | `/usr/bin/cmd/otelcol` |
+| `process.exit_code` | int | The exit code of the process. | `127` |
+| `process.group_leader.pid` | int | The PID of the process's group leader. This is also the process group ID (PGID) of the process. | `23` |
+| `process.interactive` | boolean | Whether the process is connected to an interactive shell. |  |
 | `process.owner` | string | The username of the user that owns the process. | `root` |
 | `process.parent_pid` | int | Parent Process identifier (PID). | `111` |
 | `process.pid` | int | Process identifier (PID). | `1234` |
+| `process.real_user.id` | int | The real user ID (RUID) of the process. | `1000` |
+| `process.real_user.name` | string | The username of the real user of the process. | `operator` |
 | `process.runtime.description` | string | An additional description about the runtime of the process, for example a specific vendor customization of the runtime environment. | `Eclipse OpenJ9 Eclipse OpenJ9 VM openj9-0.21.0` |
 | `process.runtime.name` | string | The name of the runtime of this process. For compiled native binaries, this SHOULD be the name of the compiler. | `OpenJDK Runtime Environment` |
 | `process.runtime.version` | string | The version of the runtime of this process, as returned by the runtime without modification. | `14.0.2` |
+| `process.saved_user.id` | int | The saved user ID (SUID) of the process. | `1002` |
+| `process.saved_user.name` | string | The username of the saved user. | `operator` |
+| `process.session_leader.pid` | int | The PID of the process's session leader. This is also the session ID (SID) of the process. | `14` |
+| `process.start` | string | The date and time the process started, in ISO 8601 format. | `2023-11-21T09:25:34.853Z` |
+| `process.user.id` | int | The effective user ID (EUID) of the process. | `1001` |
+| `process.user.name` | string | The username of the effective user of the process. | `root` |
+| `process.vpid` | int | Virtual process identifier. [2] | `12` |
+
+**[1]:** As environment variables may change during a process's lifespan, this should be captured as a snapshot when the event occurred.
+May be filtered to protect sensitive information.
```

Comment:
I think it should be then `MAY BE` or even `MUST BE`. We need to avoid sensitive information being captured

Reply:
Schema is the right place to specify these things related to the privacy/sensitive information. See for example [url schema](https://github.com/open-telemetry/semantic-conventions/blob/main/model/registry/url.yaml#L20)
Also imperative statements should be written with upper case

Reply:
I think I would vote for `SHOULD` in this case. Intuitively we know there are all kinds of things that *could* be considered sensitive information, but if we can't give a definitive list of the things that must be filtered it, then it feels weird to give it the `MUST` imperative since it's hard to be specific.

Arguably this could apply to URL as well, where the decision was to use `MUST`. So that decision could be used to inform this. In this case, sensitive information could be any manner of credentials, keys, PII, and they generally don't take any standard form. So it feels different than with URL where it seemed specific to username and password.

I don't feel too strongly, but I think I would choose `SHOULD` or `MAY` personally, unless we found a way to be more prescriptive here about exactly what type of sensitive info should be filtered out (rather than leaving it up to the implementation).

Reply:
I've changed this to `SHOULD` now

Reply:
Note that `SHOULD` actually means to avoid implementing it, there must be reasons. There was a recent discussion about the "weight" if SHOULD as it's commonly interpreted as "optional" and "not really required".  

>SHOULD   This word, or the adjective "RECOMMENDED", mean that there
   may exist valid reasons in particular circumstances to ignore a
   particular item, but the full implications must be understood and
   carefully weighed before choosing a different course.
   
https://datatracker.ietf.org/doc/html/rfc2119

Reply:
> Note that `SHOULD` actually means to avoid implementing it, there must be reasons. There was a recent discussion about the "weight" if SHOULD as it's commonly interpreted as "optional" and "not really required", which is wrong.

I feel like using `SHOULD` here matches this definition well. Given the security implications of unredacted env vars, users should understand the risks of not filtering, but still, there might be some use cases where the full data is wanted.

Reply:
I've taken `env_vars` out of this PR, and created https://github.com/open-telemetry/semantic-conventions/issues/672 to track adding it in the future, so this discussion no longer applies to this PR